bigquery/dataset.go

```diff
@@ -104,6 +125,10 @@ type DatasetMetadataToUpdate struct {
 	// in the dataset.
 	DefaultEncryptionConfig *EncryptionConfig
 
+	// Defines the default collation specification of future tables
+	// created in the dataset.
+	DefaultCollation *Collation
```

Comment:
This gets back to the "enum or string" point.  If our values are more open-ended, maybe this should be optional.String?